cdap-common/src/main/java/co/cask/cdap/common/namespace/Namespaces.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.namespace;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.Id;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Utilities for namespaces
+ */
+public class Namespaces {
+  private Namespaces() {
+  }
+
+  public static Location getNamespaceHomeLocation(LocationFactory locationFactory, CConfiguration cConf,
+                                                  Id.Namespace namespaceId) throws IOException {
+    return locationFactory.create(getNamespacesDir(cConf)).append(namespaceId.getId());
+  }
+
+  public static String getNamespacesDir(CConfiguration cConf) {
+    return cConf.get(Constants.Namespace.NAMESPACES_DIR);
+  }
+
+  public static File getLocalTemporaryDirectory(CConfiguration cConf, Id.Namespace namespaceId) {
+    String tempBase = String.format("%s/%s/%s", cConf.get(Constants.CFG_LOCAL_DATA_DIR),
```

Comment:
Don't use string format to form path as it'll break in Windows. Always use `File` to do so:

```java
File localDataDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR));
File namespaceBase = new File(localDataDir, cConf.get(Constants.Namespace.NAMESPACES_DIR));
return new File(new File(namespaceBase, namespaceId.toString()),
                         cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
```